### PR TITLE
Bump terraform providers

### PR DESCRIPTION
**Pull request by sumo-drosiek**
* Bump providers versions
 * Bump terraform client version due to bugs
 * Update terraform scripts due to deprecation warnings
 * Add one new test and update previous ones regarding aboves

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

File: deploy/helm/sumologic/templates/_helpers.tpl [hunk 1]
```diff
@@ -593,4 +593,28 @@ Example usage:
...
 {{ $endpoint }}
+{{- end -}}
+
+
+{{/*
+Add or skip quotation denending on the value
+
+Examples:
+  - "${test}" will be printed as `test`
+  - "test" will be printed as `"test"`
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/_helpers.tpl:604]: This is due to deprecation of `"${variable}"` in favor of `variable`

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -52,7 +52,7 @@ sumologic:
     # config_context_auth_info:
     # config_context_cluster:
     token: "${file(\"/var/run/secrets/kubernetes.io/serviceaccount/token\")}"
-    # load_config_file:
+    load_config_file: false
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:55]: fix for https://github.com/terraform-providers/terraform-provider-kubernetes/issues/821

**Comment by syedrakib** [comment]
Looking forward to the fix for terraform-providers/terraform-provider-kubernetes#821

**Comment by sumo-drosiek** [comment]
@syedrakib, setting `load_config_file` to `false` solves `http://localhost/api/v1/namespaces/my-ns: dial tcp 127.0.0.1:80: connect: connection refused` for us. At least I didn't spot the issue yet.

**Comment by ravitomardevops** [comment]
Does kubernets provider support load_config_file to false ? not  ,please check official page https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs ,Now how to deal with it ?

File: deploy/helm/sumologic/values.yaml (hunk 2 above)
Comment (ravitomardevops) [reply deploy/helm/sumologic/values.yaml:55]: i am also facing same issue as it is being running via github action workflow

**Comment by ravitomardevops** [comment]
> Looking forward to the fix for [hashicorp/terraform-provider-kubernetes#821](https://github.com/hashicorp/terraform-provider-kubernetes/issues/821)

@syedrakib How you have resolved it , i am also facing same issue .Please help

**Comment by sumo-drosiek** [comment]
@ravitomardevops What sumologic kubernetes collection version do you use?

**Pull request merged**